x-pack/test/cases_api_integration/security_and_spaces/tests/common/internal/bulk_create_attachments.ts

```diff
@@ -470,92 +472,457 @@ export default ({ getService }: FtrProviderContext): void => {
+            });
+
+            indices.push(alert._index);
+            ids.push(alert._id);
+          });
+
+          await bulkCreateAttachmentsAndRefreshIndex({
+            caseId: postedCase.id,
+            alerts,
+            auth: attachmentAuth,
+            expectedHttpCode: attachmentExpectedHttpCode,
+          });
+
+          const updatedAlerts = await getSecuritySolutionAlerts(supertest, ids);
+
+          updatedAlerts.hits.hits.forEach((alert) => {
+            expect(alert._source?.[ALERT_WORKFLOW_STATUS]).eql(expectedAlertStatus);
+          });
+        };
+
+        const bulkCreateAlertsAndVerifyCaseIdsInAlertSchema = async (totalCases: number) => {
+          const cases = await Promise.all(
+            [...Array(totalCases).keys()].map((index) =>
+              createCase(supertest, {
+                ...postCaseReq,
+                settings: { syncAlerts: false },
+              })
+            )
+          );
+
+          const signals = await createSecuritySolutionAlerts(supertest, log);
+          const alert = signals.hits.hits[0];
+
+          for (const theCase of cases) {
+            await bulkCreateAttachmentsAndRefreshIndex({
+              caseId: theCase.id,
+              alerts: [{ id: alert._id, index: alert._index }],
+            });
+          }
+
+          await es.indices.refresh({ index: alert._index });
+
+          const updatedAlert = await getSecuritySolutionAlerts(supertest, [alert._id]);
+          const caseIds = cases.map((theCase) => theCase.id);
+
+          expect(updatedAlert.hits.hits[0]._source?.[ALERT_CASE_IDS]).eql(caseIds);
+
+          return { updatedAlert, cases };
+        };
+
+        it('should change the status of the alerts if sync alert is on', async () => {
+          await bulkCreateAlertsAndVerifyAlertStatus({
+            syncAlerts: true,
+            expectedAlertStatus: 'acknowledged',
+          });
         });
 
-        const { id } = await createRule(supertest, log, rule);
-        await waitForRuleSuccessOrStatus(supertest, log, id);
-        await waitForSignalsToBePresent(supertest, log, 1, [id]);
-        const signals = await getSignalsByIds(supertest, log, [id]);
-        const attachments: CommentRequest[] = [];
-        const indices: string[] = [];
-        const ids: string[] = [];
-
-        signals.hits.hits.forEach((alert) => {
-          expect(alert._source?.[ALERT_WORKFLOW_STATUS]).eql('open');
-          attachments.push({
-            alertId: alert._id,
-            index: alert._index,
-            rule: {
-              id: 'id',
-              name: 'name',
+        it('should NOT change the status of the alert if sync alert is off', async () => {
+          await bulkCreateAlertsAndVerifyAlertStatus({
+            syncAlerts: false,
+            expectedAlertStatus: 'open',
+          });
+        });
+
+        it('should change the status of the alert when the user has read access only', async () => {
```

Comment:
I think this test should fail if the user only has read access to the elasticsearch index. Or we can keep it but change the test name to say something like: when the user has write access to the indices and only read access to the siem solution